python/cuda_cccl/tests/parallel/test_merge_sort.py

```diff
@@ -69,7 +73,12 @@ def test_merge_sort_keys(dtype, num_items):
 
     d_in_keys = numba.cuda.to_device(h_in_keys)
 
-    merge_sort_device(d_in_keys, None, d_in_keys, None, compare_op, num_items)
+    if dtype == np.float16:
+        operator = parallel.OpKind.LESS
+    else:
+        operator = compare_op
+
```

Comment:
Can we instead accept the `operator` as a test parameter, by extending `dtype_size_pairs` to include the operator; something like:

```python
merge_sort_params = [
    (dt, 2**log_size, parallel.OpKind.LESS if dt == np.float16 else compare_op)
    for dt in DTYPE_LIST
    for log_size in type_to_problem_sizes(dt)
]
```

Will help reduce churn if we e.g., need to do this for more data types or if we want to use custom ops when they are supported for `__half`.

Ditto everywhere else this pattern is being used.